src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -174,9 +212,16 @@ export class WorkspaceSavedObjectsClientWrapper {
       attributes: Partial<T>,
       options: SavedObjectsUpdateOptions = {}
     ): Promise<SavedObjectsUpdateResponse<T>> => {
-      if (this.isRelatedToWorkspace(type)) {
-        await this.validateSingleWorkspacePermissions(id, wrapperOptions.request, [
+      const workspacePermitted = await this.validateWorkspaceAttributesPermitted(
+        attributes,
+        wrapperOptions.request,
+        WorkspacePermissionMode.Management
+      );
+
+      if (!workspacePermitted) {
+        await this.validateSingleObjectPermissions(id, type, wrapperOptions.request, [
           WorkspacePermissionMode.Management,
+          WorkspacePermissionMode.Write,
```

Comment:
Do we need to check `WorkspacePermissionMode.LibraryWrite` when type equal WORKSPACE?